lib/_http_client.js

```diff
@@ -229,6 +229,12 @@ function ClientRequest(input, options, cb) {
   }
   this.insecureHTTPParser = insecureHTTPParser;
 
+  if (options.joinDuplicateHeaders !== undefined) {
```

Comment:
Is null an acceptable value for joinDuplicateHeaders? I think this should be != null

Reply:
There is the validate boolean function the line below. Its pretty standard